src/hevm/test/test.hs

```diff
@@ -1269,3 +1319,115 @@ bothM f (a, a') = do
 
 applyPattern :: String -> TestTree  -> TestTree
 applyPattern p = localOption (TestPattern (parseExpr p))
+
+runDappTest :: FilePath -> Text -> IO Bool
+runDappTest testFile match = do
+  root <- Paths.getDataDir
+  (json, _) <- compileWithDSTest testFile
+  --TIO.writeFile "output.json" json
+  withCurrentDirectory root $ do
+    withSystemTempFile "output.json" $ \file handle -> do
+      hClose handle
+      TIO.writeFile file json
+      withSolvers Z3 1 $ \solvers -> do
+        opts <- testOpts solvers root json match
+        dappTest opts solvers file Nothing
+
+debugDappTest :: FilePath -> IO ()
+debugDappTest testFile = do
+  root <- Paths.getDataDir
+  (json, _) <- compileWithDSTest testFile
+  --TIO.writeFile "output.json" json
+  withCurrentDirectory root $ do
+    withSystemTempFile "output.json" $ \file handle -> do
+      hClose handle
+      TIO.writeFile file json
+      withSolvers Z3 1 $ \solvers -> do
+        opts <- testOpts solvers root json ".*"
+        TTY.main opts root file
+
+
+testOpts :: SolverGroup -> FilePath -> Text -> Text -> IO UnitTestOptions
+testOpts solvers root solcJson match = do
+  srcInfo <- case readJSON solcJson of
+               Nothing -> error "Could not read solc json"
+               Just (contractMap, asts, sources) -> do
+                 sourceCache <- makeSourceCache sources asts
+                 pure $ dappInfo root contractMap sourceCache
+
+  params <- getParametersFromEnvironmentVariables Nothing
+
+  pure EVM.UnitTest.UnitTestOptions
+    { EVM.UnitTest.oracle = Fetch.oracle solvers Nothing
+    , EVM.UnitTest.maxIter = Nothing
+    , EVM.UnitTest.askSmtIters = Nothing
+    , EVM.UnitTest.smtTimeout = Nothing
+    , EVM.UnitTest.solver = Nothing
+    , EVM.UnitTest.covMatch = Nothing
+    , EVM.UnitTest.verbose = Nothing
+    , EVM.UnitTest.match = match
+    , EVM.UnitTest.maxDepth = Nothing
+    , EVM.UnitTest.fuzzRuns = 100
+    , EVM.UnitTest.replay = Nothing
+    , EVM.UnitTest.vmModifier = id
+    , EVM.UnitTest.testParams = params
+    , EVM.UnitTest.dapp = srcInfo
+    , EVM.UnitTest.ffiAllowed = True
+    }
+
+compileWithDSTest :: FilePath -> IO (Text, Text)
+compileWithDSTest src =
+  withSystemTempFile "input.json" $ \file handle -> do
+    hClose handle
+    dsTest <- readFile =<< Paths.getDataFileName "test/contracts/lib/test.sol"
```

Comment:
right now we use a vendored version of ds-test. It would be nice if we could take a path via env var or something that would let us test against an arbitrary ds-test commit...